File: ABC208/C.rb
n, k = *gets.split(" ")
n = n.to_i
k = k.to_i

aaa = gets.split(" ").map{|a| a.to_i}
base, bonus = *k.divmod(n)

sorted_aaa = aaa.sort
th = bonus == 0 ? -1 : sorted_aaa[bonus-1].to_i

for a in aaa do
  if a.to_i <= th
    puts base + 1
  else
    puts base
  end
end

File: ABC208/C.py
N, K = map(int,input().split())
aList = [int(x) for x in input().split()]

base = K // N
bonus = K % N

sorted_a = sorted(aList)
th = sorted_a[bonus - 1] if bonus > 0 else -1

for a in aList:
    if a <= th:
        print(base + 1)
    else:
        print(base)